src/HeatmapView.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import MarkerGeneHeatmap from './MarkerGeneHeatmap'
+import PlotSettingsDropdown from './PlotSettingsDropdown'
+import LoadingOverlay from './LoadingOverlay'
+import CalloutAlert from './CalloutAlert'
+
+import URI from 'urijs'
+import _ from 'lodash'
+
+class HeatmapView extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      data: [],
+      filteredData: [],
+      selectedClusterId: null,
+      isLoading: true,
+      hasError: null
+    }
+  }
+
+  async _fetchAndSetState({resource, host}) {
+    this.setState({
+      isLoading: true
+    })
+
+    const url = URI(resource, host).toString()
+
+    try {
+      const response = await fetch(url)
+
+      if (!response.ok) {
+        throw new Error(`${url} => ${response.status}`)
+      }
+
+      const jsonData = await response.json()
+      this.setState({
+        data: await jsonData,
+        filteredData: await jsonData,
+        selectedClusterId: null,
+        isLoading: false,
+        hasError: null
+      })
+    } catch(e) {
+      this.setState({
+        data: null,
+        isLoading: false,
+        hasError: {
+          description: `There was a problem communicating with the server. Please try again later.`,
+          name: e.name,
+          message: e.message
+        }
+      })
+    }
+  }
+
+  componentDidMount() {
+    this._fetchAndSetState(this.props)
+  }
+
+  componentDidUpdate(previousProps) {
+    if (previousProps.resource !== this.props.resource) {
+      this._fetchAndSetState(this.props)
+    }
+  }
+
+  componentDidCatch(error, info) {
+    this.setState({
+      hasError: {
+        description: `There was a problem rendering this component.`,
+        name: error.name,
+        message: `${error.message} – ${info}`
+      }
+    })
+  }
+
+  render() {
+    const { data, filteredData, selectedClusterId, isLoading, hasError } = this.state
+    const { wrapperClassName, plotWrapperClassName } = this.props
+    const { ks, ksWithMarkers, selectedK, onSelectK } = this.props
+    const { hasDynamicHeight, defaultHeatmapHeight, heatmapRowHeight } = this.props
+
+    const kOptions = ks.sort((a, b) => a-b).map((k) => ({
+      value: k.toString(),
+      label: `k = ${k}`,
+      isDisabled: ksWithMarkers ? !ksWithMarkers.includes(k) : false
+    }))
+
+    const allClusterIds = Array.from(Array(Number.parseInt(selectedK)+1).keys()).slice(1)
```

Comment:
`parseInt` is global, there’s no need to qualify it with `Number`. Also, since we’re importing `lodash`, let’s use it:
`_.range(1, parseInt(selectedK) + 1)`